src/ompl/base/StateSampler.h

```diff
@@ -76,6 +76,26 @@ namespace ompl
             {
             }
 
+            /** \brief Cast this instance to a desired type. */
+            template<class T>
```

Comment:
This encourages users to cast samplers to other types. In general, the samplers in OMPL are really basic things, there should ideally be no need to ever cast them